cfme/infrastructure/provider/openstack_infra.py

```diff
@@ -55,31 +41,133 @@ def has_nodes(self):
 
     @classmethod
     def from_config(cls, prov_config, prov_key, appliance=None):
-        credentials_key = prov_config['credentials']
-        credentials = cls.process_credential_yaml_key(credentials_key)
-        credential_dict = {'default': credentials}
+        endpoints = [RHOSEndpoint(**prov_config['endpoints']['default']),
```

Comment:
@psav, I don't see any way to handle missing credential somehow or other. So, I'd prefer to just throw error. Here we will have error as early as it is possible If some credential is missing. It will happen on creating endpoint. 
It is possible that I just don't follow you. So, I'd be glad to discuss that in chat.